### DISCUSSION TITLE
migrate git lfs to elsewhere

### DISCUSSION THREAD

**Discussion by laoshaw** [git-lfs/git-lfs#5512] [General]
We're thinking about adopting LFS in our git repos. However it's unclear to me yet, that say if I have a LFS at github or gitlab or gitea, and we decided to go with someone else later, can I migrate all existing LFS histories and files to another vendor easily?

By easily I mean, I do a git-lfs-pull, change upstream repo URL, then git-lfs-push, and now my LFS is working with a different repo at somewhere else(a different vendor, for example).

Is this an easy process or very complex and involved process in practice?

**Comment by bk2204** [comment]
It's possible to move all of the files from one location to the other without a problem.  You can do `git lfs fetch --all`, and then `git lfs push --all NEW-REMOTE` (where `NEW-REMOTE` is the new remote).

If you're using `lfs.url`, then things become more complicated as you check out older versions, since `lfs.url` will be wrong, and you'll need to override it manually in the local `.git/config` for that to work.  Newer versions with an up-to-date `lfs.url` should be fine.  However, if you're not doing that and you're just using the default remote (which might point somewhere else in the future as you change forges), then it's just a full fetch and a full push.

**Reply by laoshaw** [reply]
Thanks a lot! That helps me to promote lfs usage in the company. We plan to self-host gitlab and are wondering in the unlikely case if we will switch vendors later(e.g. from gitlab to github or gitea, or whatever), will LFS be able to switch while keeping all its commit history info. My understanding is that as long as LFS server is following the standard APIs, it should be portable to avoid vendor-lockin.

**Reply by bk2204** [reply]
Yup, that's totally possible.  Git LFS has a standard set of APIs which anyone is welcome to implement. I know GitLab and GitHub implement them, and I'm 90% sure that Gitea does as well.
